EXTENSIONS.md

```diff
@@ -0,0 +1,236 @@
+
+This example can be desugared as
+
+```js
+let decInstance = dec(undefined, {kind: "parameter"});
+
+function f(arg_) {
+  const arg = decInstance(arg_);
+  return arg * 2 ;
+}
+
+f[Symbol.annotations] = {}
+f[Symbol.annotations].parameters = []
+f[Symbol.annotations].parameters[0] = {x: "y"};
+```
+
+## `let` decorators
+
+Variables declared with `let` can be decorated, converting them into special getter/setter pairs that are invoked when the variable is read or written.
+
+```js
+let @deprecated x = 1;
+++x;  // Shows deprecation warnings for read and write
+```
+
+`let` decorators might be useful for systems using reactivity based on local variables, e.g. "hooks" systems.
+
+`let` decorator details:
+- First parameter: A `{ get, set, value }` object (where the receiver is expected to be undefined, and where `value` is the RHS)
+- Second parameter: a context object `{ kind: "let" }`
+- Return value: A `{ get, set, value }` object, with potentially new behavior
+
+This example can be desugared as:
+
+```js
+let { get_x, set_x, value: _x } = deprecated({value: 1, get() { return _x; }, set(v) { _x = v; }}, {kind: "let"});
+
+set_x(get_x()++);
+```
+
+## `const` decorators
+
+Variables declared with `const` can be decorated more simply--the decorator simply wraps the value being decorated when it's being defined.
+
+```js
+function inc(x) { return x+1; }
+const @inc x = 1;  // 2
+```
+
+`const` decorator details:
+- First parameter: The value of the RHS
+- Second parameter: a context object `{ kind: "const" }`
+- Return value: A new value for the variable
+
+This could be desugared as follows:
+
+```js
+const x = inc(1, {kind: "const"});
+```
+
+(This form isn't so useful by itself, but may become more important if future proposals share more information through the context object.)
+
+## Object literal and property decorators and annotations
+
+- A decorated object literal works like a class decorator, but with `kind: "object"`.
+- A decorated method, getter or setter in an object literal works just like one in a class, replacing that method.
+- A decorated object property works like a field decorator, but with `kind: "property"`, and it receives the initial value in the `value` property of the input, and returns it in the output object, rather than returning an initializer function, since it only runs once (in this way, it is similar to `let` decorators).
+
+Example:
+
+```js
+const x = @decA {
+  @decB p: v,
+  @decC m() { }
+};
+```
+
+<!--
+would desugar roughly into (in terms of the [`private`/`with` proposal](https://gist.github.com/littledan/5451d6426a8ed65c0f3c2822c51314d1))
+
+```js
+private #p;
+let {get, set, value} = decB() // tood: fihish)
+
+const x = decA({
+  #p: initializer(value)
+  valuedec
+  m: decC(function m() { }, {kind: "method"})
+}, {kind: "object"});
+```
+-->
+
+## Block decorators
+
+Decorating a block could wrap it as a function.
+
+```js
+@foo { bar; }
```

Comment:
Does it become an expression after being decorated?

```mjs
let x = await @some_promise_transformation {
    await expr;
}
```

Reply:
No, that would not parse as a decorated block, but rather a decorated object literal, unfortunately. I think you might want an async do expression (built-in syntax, not a decorator) for this use case.

Reply:
What about this?

```js
async function foo() {
	@decorator { await expr; }
}
```

Would the inner block be in an `async` context, too? And if so, what would the completion value be?